web/src/features/fba/pages/FireBehaviourAdvisoryPage.tsx

```diff
@@ -152,17 +154,49 @@ const FireBehaviourAdvisoryPage: React.FunctionComponent = () => {
     document.title = ASA_DOC_TITLE
   }, [])
 
+  const formControlRef = useRef<HTMLDivElement>(null)
+  const mapRef = useRef<HTMLDivElement>(null)
+  const navRef = useRef<HTMLDivElement>(null)
+  const sidePanelRef = useRef<HTMLDivElement>(null)
+  const [formControlHeight, setFormControlHeight] = useState<number>(0)
+  const [navRefHeight, setNavRefHeight] = useState<number>(0)
+
+  useEffect(() => {
+    if (navRef.current) {
+      setNavRefHeight(navRef.current.clientHeight)
+    }
+  }, [navRef.current?.clientHeight])
+
+  useEffect(() => {
+    if (formControlRef.current) {
+      setFormControlHeight(formControlRef.current.clientHeight)
+    }
+  }, [formControlRef.current?.clientHeight])
+
+  useEffect(() => {
+    const sidePanelElement = sidePanelRef.current
+    const mapElement = mapRef.current
+    if (sidePanelElement && mapElement && formControlHeight && navRefHeight) {
+      const height = `calc(100vh - ${formControlHeight + navRefHeight}px)`
+      sidePanelElement.style.height = height
```

Comment:
This works great, but there is a more 'Reactful' way of setting the height of the map and side panel elements. I think setting properties on components by directly accessing the dom attributes is a bit of an anti-pattern in React, though perhaps it's just stylistic. I'll leave the choice of how you do this up to you, but I just wanted to point out another option since you're still relatively new to React. See my comments below which would allow you to get rid of this useEffect hook, the mapRef and the sidePanelRef.